pkg/operator/controllers/checkers/ingresscertificatechecker/checker_test.go

```diff
@@ -11,81 +11,167 @@ import (
 		},
-		Spec: configv1.ClusterVersionSpec{
-			ClusterID: configv1.ClusterID("fake-cluster-id"),
+		{
+			name:              "Check returns IngressController not found error when IngressController is nil",
+			clusterVersion:    fakeClusterVersion(),
+			ingressController: nil,
+			wantErr:           `ingresscontrollers.operator.openshift.io "default" not found`,
+		},
+		{
+			name:              "Check returns Cluster not found error when Cluster is nil",
+			cluster:           nil,
+			clusterVersion:    fakeClusterVersion(),
+			ingressController: fakeIngressController(nil),
+			wantErr:           "clusters.aro.openshift.io \"cluster\" not found",
+		},
+		{
+			name:              "Check returns certificate error (ARO's fault) when no certificate is set and cluster has a managed domain",
+			clusterVersion:    fakeClusterVersion(),
+			ingressController: fakeIngressController(nil),
+			cluster:           fakeCluster(managedDomain),
+			wantErr:           errNoCertificateAndManagedDomain.Error(),
+		},
+		{
+			name:              "Check returns error (customer's fault) when no certificate is set and cluster has a custom domain",
+			clusterVersion:    fakeClusterVersion(),
+			ingressController: fakeIngressController(nil),
+			cluster:           fakeCluster(customDomain),
+			wantErr:           errNoCertificateAndCustomDomain.Error(),
+		},
+		{
+			name:           "Check returns error when cluster has a managed domain and there is an invalid certificate name",
+			clusterVersion: fakeClusterVersion(),
+			ingressController: fakeIngressController(&corev1.LocalObjectReference{
+				Name: "fake-custom-name-ingress",
+			}),
+			cluster: fakeCluster(managedDomain),
+			wantErr: errInvalidCertificateAndManagedDomain.Error(),
 		},
-	}
-
-	for _, tt := range []struct {
-		name              string
-		ingressController *operatorv1.IngressController
-		clusterVersion    *configv1.ClusterVersion
-		wantErr           string
-	}{
 		{
-			name:           "default certificate is set",
-			clusterVersion: clusterVersion,
+			name:           "Check returns no error when cluster has a managed domain and certificate has a valid name",
+			clusterVersion: fakeClusterVersion(),
 			ingressController: fakeIngressController(&corev1.LocalObjectReference{
 				Name: "fake-cluster-id-ingress",
 			}),
+			cluster: fakeCluster(managedDomain),
 		},
 		{
-			name:           "unexpected certificate name",
-			clusterVersion: clusterVersion,
+			name:           "Check returns error when cluster has a managed domain and certificate name is empty",
+			clusterVersion: fakeClusterVersion(),
 			ingressController: fakeIngressController(&corev1.LocalObjectReference{
-				Name: "fake-custom-name-ingress",
+				Name: "",
 			}),
-			wantErr: `custom ingress certificate in use: "fake-custom-name-ingress"`,
+			cluster: fakeCluster(managedDomain),
+			wantErr: "invalid ingress certificate name for cluster with managed domain",
```

Comment:
For consistency with the other tests, can we use the custom error here instead of a string?

```suggestion
			wantErr: errInvalidCertificateAndManagedDomain.Error(),
```

Reply:
Yes, I forgot to change it. 
Thanks!
Done.